src/map/mob_spell_container.cpp

```diff
@@ -157,6 +157,73 @@ std::optional<SpellID> CMobSpellContainer::GetBestAvailable(SPELLFAMILY family)
     return (!matches.empty()) ? std::optional<SpellID>{ matches.back() } : std::nullopt;
 }
 
+std::optional<SpellID> CMobSpellContainer::GetBestAgainstTargetWeakness(CBattleEntity* PTarget)
+{
+    // Look up what the target has the _least resistance to_:
+    std::vector<int16> resistances
+    {
+         PTarget->getMod(Mod::FIRERES),
+         PTarget->getMod(Mod::ICERES),
+         PTarget->getMod(Mod::WINDRES),
+         PTarget->getMod(Mod::EARTHRES),
+         PTarget->getMod(Mod::THUNDERRES),
+         PTarget->getMod(Mod::WATERRES),
+         PTarget->getMod(Mod::LIGHTRES),
+         PTarget->getMod(Mod::DARKRES),
+    };
+
+    std::size_t weakestIndex = std::distance(resistances.begin(), std::min_element(resistances.begin(), resistances.end()));
+
+    // TODO: Figure this out properly:
+    std::optional<SpellID> choice = std::nullopt;
+    switch (weakestIndex + 1) // Adjust to ignore ELEMENT_NONE
+    {
+        case ELEMENT_FIRE:
+        {
+            choice = GetBestAvailable(SPELLFAMILY_FIRE);
+            break;
+        }
+        case ELEMENT_ICE:
+        {
+            choice = GetBestAvailable(SPELLFAMILY_BLIZZARD);
+            break;
+        }
+        case ELEMENT_WIND:
+        {
+            choice = GetBestAvailable(SPELLFAMILY_AERO);
+            break;
+        }
+        case ELEMENT_EARTH:
+        {
+            choice = GetBestAvailable(SPELLFAMILY_STONE);
+            break;
+        }
+        case ELEMENT_THUNDER:
+        {
+            choice = GetBestAvailable(SPELLFAMILY_THUNDER);
+            break;
+        }
+        case ELEMENT_WATER:
+        {
+            choice = GetBestAvailable(SPELLFAMILY_WATER);
+            break;
+        }
+        case ELEMENT_LIGHT:
+        {
+            choice = GetBestAvailable(SPELLFAMILY_BANISH);
+            break;
+        }
+        case ELEMENT_DARK:
+        {
+            choice = GetBestAvailable(SPELLFAMILY_DRAIN);
+            break;
+        }
+    }
+
+    // If all else fails, just cast the best you have!
+    return !choice ? GetBestAvailable(SPELLFAMILY_NONE) : choice;
+}
+
```

Comment:
will this make the mobs attack the players weakest elemental resistance or it only applies to attacks to mobs?

Reply:
This is currently only used for Trusts, we're trying to figure out if there are any actual mobs that need really smart AI or not

Reply:
awesome thanks =)

I'll try doing some campaign battle and intentionally lower my magic resistance and see if mobs tend to select that element more. would be good research ❤️

Reply:
> This is currently only used for Trusts, we're trying to figure out if there are any actual mobs that need really smart AI or not

def were a couple back in 75 cap. my best example has sadly been dumbed down by SE and I don't know why.
https://ffxiclopedia.fandom.com/wiki/Phanduron_the_Condemned
> Rather high casting intelligence compared to most caster mobs.

they weren't talking about his INT stat. he'd hang onto stun till you were trying to do something he didn't want you doing, as just one example. If stun was on cooldown he'd sleep you instead just to interrupt you. Brought a healer? your healer is getting slept. Party sleepga'd. Shadows? he'll wait for you to just finish recasting then strip those immediately. That was all back before they moved his pos and added lv 80- mobs to the zone. Now in modern ffxi he's a a dumb as bricks blm that selects spells at random even when it won't help him at all. :cry: